workflow/controller/controller.go

```diff
@@ -184,6 +195,45 @@ func (wfc *WorkflowController) Run(ctx context.Context, wfWorkers, podWorkers in
 	<-ctx.Done()
 }
 
+// Create and initialize the Synchronization Manager
+func (wfc *WorkflowController) createSynchronizationManager() error {
+	syncLimitConfig := func(lockKey string) (int, error) {
+		lockName, err := sync.DecodeLockName(lockKey)
+		if err != nil {
+			return 0, err
+		}
+		configMap, err := wfc.kubeclientset.CoreV1().ConfigMaps(lockName.Namespace).Get(lockName.ResourceName, metav1.GetOptions{})
+		if err != nil {
+			return 0, err
+		}
+
+		value, found := configMap.Data[lockName.Key]
+		if !found {
+			return 0, argoErr.New(argoErr.CodeBadRequest, "Invalid Sync configuration Key")
+		}
+		return strconv.Atoi(value)
+	}
+
+	wfc.syncManager = sync.NewLockManager(syncLimitConfig, func(key string) {
+		wfc.wfQueue.AddAfter(key, 0)
+	})
+
+	labelSelector := v1Label.NewSelector()
+	req, _ := v1Label.NewRequirement(common.LabelKeyPhase, selection.Equals, []string{string(wfv1.NodeRunning)})
+	if req != nil {
+		labelSelector = labelSelector.Add(*req)
+	}
+
+	listOpts := metav1.ListOptions{LabelSelector: labelSelector.String()}
+	wfList, err := wfc.wfclientset.ArgoprojV1alpha1().Workflows(wfc.namespace).List(listOpts)
+	if err != nil {
+		return err
+	}
+
+	wfc.syncManager.Initialize(wfList)
+	return err
```

Comment:
This should be 
```suggestion
	return nil
```

Since `err` is already found to be `!= nil`